### ISSUE TITLE
Panic in 404.html with Hugo v0.65.0

### ISSUE THREAD

**Issue by zoni** [google/docsy#210]
Docsy appears to crash with Hugo v0.65.0:

```
Error: Error building site: render of "404" failed: execute of template failed: template: 404.html:8:9: executing "404.html" at <partial "navbar.html" .>: error calling partial: "/tmp/docsy/layouts/partials/navbar.html:13:31": execute of template failed: template: partials/navbar.html:13:31: executing "partials/navbar.html" at <$.IsDescendant>: error calling IsDescendant: runtime error: invalid memory address or nil pointer dereference
```

This looks to be reproducible when trying to serve docsy's own user guide:

```
$  git clone --recurse-submodules --depth 1 https://github.com/google/docsy.git
Cloning into 'docsy'...
remote: Enumerating objects: 220, done.
remote: Counting objects: 100% (220/220), done.
remote: Compressing objects: 100% (202/202), done.
remote: Total 220 (delta 12), reused 142 (delta 4), pack-reused 0
Receiving objects: 100% (220/220), 3.29 MiB | 3.98 MiB/s, done.
Resolving deltas: 100% (12/12), done.
Submodule 'assets/vendor/Font-Awesome' (https://github.com/FortAwesome/Font-Awesome.git) registered for path 'assets/vendor/Font-Awesome'
Submodule 'assets/vendor/bootstrap' (https://github.com/twbs/bootstrap.git) registered for path 'assets/vendor/bootstrap'
Cloning into '/tmp/docsy/assets/vendor/Font-Awesome'...
remote: Enumerating objects: 108500, done.        
remote: Total 108500 (delta 0), reused 0 (delta 0), pack-reused 108500        
Receiving objects: 100% (108500/108500), 90.98 MiB | 7.45 MiB/s, done.
Resolving deltas: 100% (63186/63186), done.
Cloning into '/tmp/docsy/assets/vendor/bootstrap'...
remote: Enumerating objects: 47, done.        
remote: Counting objects: 100% (47/47), done.        
remote: Compressing objects: 100% (38/38), done.        
remote: Total 147631 (delta 18), reused 17 (delta 8), pack-reused 147584        
Receiving objects: 100% (147631/147631), 149.24 MiB | 6.79 MiB/s, done.
Resolving deltas: 100% (97056/97056), done.
Submodule path 'assets/vendor/Font-Awesome': checked out '538a5bc4cf7989821d052e3ff919e2bead07021c'
Submodule path 'assets/vendor/bootstrap': checked out '3b558734382ce58b51e5fc676453bfd53bba9201'

$ cd docsy/userguide

$  hugo version
Hugo Static Site Generator v0.65.0/extended linux/amd64 BuildDate: unknown

$  hugo server --themesDir ../..                                               
Built in 874 ms
Error: Error building site: render of "404" failed: execute of template failed: template: 404.html:8:9: executing "404.html" at <partial "navbar.html" .>: error calling partial: "/tmp/docsy/layouts/partials/navbar.html:13:31": execute of template failed: template: partials/navbar.html:13:31: executing "partials/navbar.html" at <$.IsDescendant>: error calling IsDescendant: runtime error: invalid memory address or nil pointer dereference
```

**Comment by tshort** [comment]
Same here...

**Comment by tshort** [comment]
Probably related:

https://discourse.gohugo.io/t/runtime-error-calling-functions-on-section/23581

https://github.com/gohugoio/hugo/issues/6931

**Comment by martignoni** [comment]
Confirming with Hugo 0.65.2.

**Comment by zoni** [comment]
This is indeed https://github.com/gohugoio/hugo/issues/6931 as @tshort already remarked on above.

It's also on Hugo's radar and said to be fixed with 0.65.3, likely somewhere next week:

> See https://github.com/gohugoio/hugo/issues/6931
> 
> I’m sorry about this breakage, but it’s all for a greater good. I fixed the .Parent vs 404 and other such “standalone pages”, but I didn’t check the other related methods (.IsAncestors etc.) Sadly we did not have any tests covering it and none of the many sites I tested with caught this issue.
> 
> I will fix this early next week.

_Source: https://discourse.gohugo.io/t/runtime-error-calling-functions-on-section/23581/2_

Even though this is an upstream bug, I think it makes sense to keep this open for visibility until it's been resolved by Hugo's maintainers.

**Comment by zoni** [comment]
Also, in case anyone is looking for a workaround for this issue (other than downgrading to Hugo < 0.65.x) a simple way would be to overwrite the 404 template with something which doesn't call Docsy's layouts.

For example, create a `layouts/404.html` (from the root of your site directory) with just `404 page not found` in it.

**Comment by martignoni** [comment]
https://github.com/gohugoio/hugo/issues/6931 is closed. The fix is available in [Hugo v0.65.3](https://gohugo.io/news/0.65.3-relnotes/).

**Comment by zoni** [comment]
Indeed. :tada: With that being fixed and released, I think it's safe to close this issue now.
